libfreerdp/core/activation.c

```diff
@@ -497,39 +520,17 @@ BOOL rdp_recv_font_map_pdu(rdpRdp* rdp, wStream* s)
 	WINPR_ASSERT(rdp);
 	WINPR_ASSERT(rdp->settings);
 	WINPR_ASSERT(s);
+	WINPR_ASSERT(!freerdp_settings_get_bool(rdp->settings, FreeRDP_ServerMode));
 
-	if (rdp->settings->ServerMode)
-		return rdp_recv_server_font_map_pdu(rdp, s);
-	else
-		return rdp_recv_client_font_map_pdu(rdp, s);
-}
-
-BOOL rdp_recv_server_font_map_pdu(rdpRdp* rdp, wStream* s)
-{
-	WINPR_ASSERT(rdp);
-	WINPR_ASSERT(s);
-
-	WLog_WARN(TAG, "Invalid PDU received: FONT_MAP only allowed client -> server");
-	return rdp_finalize_set_flag(rdp, FINALIZE_SC_FONT_MAP_PDU);
-}
-
-BOOL rdp_recv_client_font_map_pdu(rdpRdp* rdp, wStream* s)
```

Comment:
Here the behaviour of function is changed, it used to support a truncated PDU (I think it happens sometime) and with your changes it would be rejected with an error.